src/file-engine/src/manifest.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use common_datasource::file_format::Format;
+use object_store::ObjectStore;
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, OptionExt, ResultExt};
+use store_api::metadata::{ColumnMetadata, RegionMetadataBuilder, RegionMetadataRef};
+use store_api::storage::{ColumnId, RegionId};
+
+use crate::error::{
+    CheckObjectSnafu, DecodeJsonSnafu, DeleteRegionManifestSnafu, EncodeJsonSnafu,
+    InvalidMetadataSnafu, LoadRegionManifestSnafu, ManifestExistsSnafu, MissingRequiredFieldSnafu,
+    ParseFileFormatSnafu, Result, StoreRegionManifestSnafu,
+};
+use crate::FileOptions;
+
+#[inline]
+fn region_manifest_path(region_dir: &str) -> String {
+    format!("{region_dir}manifest/_immutable_manifest")
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct FileRegionManifest {
+    pub region_id: RegionId,
+    pub column_metadatas: Vec<ColumnMetadata>,
+    pub primary_key: Vec<ColumnId>,
```

Comment:
Does file engine need primary keys? cc @WenyXu

Reply:
No

Reply:
So we can ignore the time index and other indexes for file engine?

Reply:
> So we can ignore the time index and other indexes for file engine?

Yes, We can